js/views/modals/Settings/Settings.js

```diff
@@ -0,0 +1,85 @@
+import loadTemplate from '../../../utils/loadTemplate';
+import BaseModal from '../BaseModal';
+import SettingsGeneral from './SettingsGeneral';
+import SettingsPage from './SettingsPage';
+import $ from 'jquery';
+
+export default class extends BaseModal {
+  constructor(options = {}) {
+    const opts = {
+      removeOnClose: true,
+      modelContentClass: 'modalContent clrP border clrBr',
+      ...options,
+    };
+
+    super(opts);
+    this.options = opts;
+
+    this.currentTabView = '';
+    this.TabViews = { SettingsGeneral, SettingsPage };
+
+    if (this.options.removeOnClose) this.on('close', () => this.remove());
+  }
+
+  className() {
+    return `${super.className()} settings`;
+  }
+
+  events() {
+    return {
+      'click .js-tab': 'tabClick',
+      'click .js-cancel': 'cancelClick',
+      'click .js-save': 'saveClick',
+      ...super.events(),
+    };
+  }
+
+  tabClick(e) {
+    const targ = $(e.target).closest('.js-tab');
+    this.selectTab(targ);
+  }
+
+  selectTab(targ) {
+    const targTab = targ.data('tab');
+    const TabView = `Settings${targTab}`;
+
+    this.$('.js-tab').removeClass('clrT active');
+    targ.addClass('clrT active');
+
+    if (!this[TabView] && this.TabViews[TabView]) {
```

Comment:
Code I didn't finish. This was supposed to only create a new view if it hadn't been already created.